ggshield/core/scan/commit_utils.py

```diff
@@ -79,13 +83,14 @@ def path(self) -> Path:
         return self._path
 
     def is_longer_than(self, max_utf8_encoded_size: int) -> bool:
-        if self._utf8_encoded_size is None:
-            self._utf8_encoded_size = len(self._content.encode(errors="replace"))
-        return self._utf8_encoded_size > max_utf8_encoded_size
-
-    @property
-    def content(self) -> str:
-        return self._content
+        self._read_content()
+        if (
+            self._utf8_encoded_size is not None
+        ):  # it should not be, but just in case for typing
+            return self._utf8_encoded_size > max_utf8_encoded_size
+        raise ValueError(
+            f"For file {self.path}, _utf8_encoded_size is None after reading."
+        )
```

Comment:
As you said: "it should not be", so the check can be done using an assert at this point (it's a programmer error if it fails). It is enough to make pyright happy:

```suggestion
        self._read_content()
        assert self._utf8_encoded_size is not None
        return self._utf8_encoded_size > max_utf8_encoded_size
```